strawberryfields/ops.py

```diff
@@ -1453,16 +1415,20 @@ def __init__(self, s=1):
 
     def _decompose(self, reg, **kwargs):
         s = self.p[0]
-        r = arcsinh(-s/2)
-        theta = 0.5*arctan2(-1.0/cosh(r), -tanh(r))
+        r = pf.asinh(-s/2)
+        #theta = 0.5 * pf.atan2(-1.0 / pf.cosh(r), -pf.tanh(r))
+        # FIXME in sympy 1.4 atan2._eval_evalf() has a bug, it does not work with Symbol._eval_evalf().
+        # This is a workaround. When sympy is fixed (in version 1.5?), go back to using pf.atan2.
+        # See https://github.com/sympy/sympy/pull/17469
+        # If s<0 we need to add pi/2 to theta. If s==0, we need to avoid division by zero.
+        temp = 0.5 * pf.atan(1 / pf.sinh(r))  # NOTE s==0 will cause a division by zero when this is evaluated
+        theta = temp +pf.Heaviside(-s) * np.pi/2
```

Comment:
@smite, @co9olguy. Is the above correct? When I plot line 1419 vs line 1424-1425, I get different results for s<0:

![Screenshot_20191011_171531](https://user-images.githubusercontent.com/2959003/66685495-bcc0d580-ec4a-11e9-8435-863793f87ef5.png)

Reply:
changing the sign of the Heaviside function leads to the correct result.
```suggestion
        theta = temp - pf.Heaviside(-s) * np.pi/2
```

Reply:
![Screenshot_20191011_171741](https://user-images.githubusercontent.com/2959003/66685612-0b6e6f80-ec4b-11e9-9e8f-87d3f60b41c8.png)

Reply:
@nquesada

Reply:
`theta` is only defined up to `n \pi` in the CXgate decomposition formula so I think in this case there is no practical difference between the two expressions.

Reply:
How come this was switched away from `piecewise`? According to the comment in code, we will get a division by zero now when s=0, whereas before we were accounting for it

Reply:
Currently Piecewise is implemented in a bad way in Sympy, it evaluates every expression regardless of the truth value of the conditions, and also causes a divide by zero error. It also does not work with batched input because `parameters.wrap_mathfunc` can't handle the more complicated way it treats its arguments.

Reply:
If the divide by zero error is happening for a case where the gate would have a parameter of 0 (i.e., identity), can we not catch that earlier with simple python logic and avoid the decomposition entirely?

Reply:
No, because at the point where the decomposition is made the quantities are still symbolic, we do not know their numeric values. But this entire issue should go away when atan2 is fixed in the next Sympy version.